src/hotspot/share/cds/filemap.cpp

```diff
@@ -1099,6 +1125,12 @@ class FileHeaderHelper {
       return false;
     }
 
+    if (gen_header._version !=  CURRENT_CDS_ARCHIVE_VERSION) {
+      auto warning_continue = FileMapInfo::fail_continue;
+      warning_continue("The shared archive file version %d which is not current version %d",
+                                    gen_header._version, CURRENT_CDS_ARCHIVE_VERSION);
+    }
```

Comment:
There's no need to add `warning_continue`. `FileMapInfo::fail_continue` means:

- when running with -Xshare:auto, print a warning and continue
- when running with -Xshare:on, print an error and exit

All other calls of `FileMapInfo::fail_continue` expects this behavior.